packages/atoms/connect/ConnectButton.tsx

```diff
@@ -13,44 +13,40 @@ import type { AtomsGlobalConfigProps } from "../types";
 interface ConnectButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
   buttonText?: string;
   icon?: JSX.Element;
-  handleClick: () => void;
   onSuccess?: () => void;
   onError: () => void;
 }
 
 export function ConnectButton({
   buttonText,
-  handleClick,
+  onClick,
   onSuccess,
   onError,
   className,
   icon,
 }: ConnectButtonProps & AtomsGlobalConfigProps) {
   const [isProcessing, setIsProcessing] = useState<boolean>(false);
-  const [isSuccess, setIsSuccess] = useState<boolean>(false);
   const [errMsg, setErrMsg] = useState<string>("");
 
   function handleSubmit(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
     e.preventDefault();
 
     try {
       setIsProcessing(true);
-      setIsSuccess(true);
-      handleClick();
+      onClick && onClick(e);
```

Comment:
onClick can be possibly undefined so checking first to see if there's a method being passed